### PR TITLE
Add a header comment to generated files.

### PR DESCRIPTION
**Pull request by bonds0097**
Add a header comment to all generated files that explains that the code was generated by this package and lists the Avro schema source files that this was built from.

The idea behind this is that it is helpful for code reviews and such to be able to distinguish generated code from hand-written code.

### CODE REVIEW

**Comment by alanctgardner** [comment]
I like the idea of adding a comment to each generated file. I've been working on making the `generator` package as generic as possible, so it could be re-used for other code generators. In that vein, I'd rather have a method on `generator.Package` like `AddFileComment(file string, commentText string)`, which would insert a new multi-line comment at the top of the file. Then we'd have more flexibility in what goes into the comment body.

**Comment by bonds0097** [comment]
Ok, followed your suggested pattern by implementing a generic AddHeader method that adds headers to files and added the actual logic of adding the codegen comment text to `main`. Let me know if that works.

**Comment by bonds0097** [comment]
Test failed, so I'll get that fixed.  I wish go had a better way of dealing with forked repos.

**Comment by bonds0097** [comment]
Fixed the tests (my code was bad, oops) so should be all good now.

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * SOURCE:
+ *     file_1.avsc
+ */`},
+	}
+
+	for i, tc := range tt {
+		got := codegenComment(tc.in)
+		expected := tc.out
+
+		if got != expected {
```
Comment (alanctgardner) [inline main_test.go]: I've been using [testify](https://github.com/stretchr/testify) for these kind of assertions, but it's not a big deal here.

Comment (bonds0097) [inline main_test.go]: Ah, I'll do that to maintain consistency.

**Comment by alanctgardner** [comment]
Thanks for the quick change, I'll merge this once CI passes!

**Pull request merged**